elastic/kibana
# [Response Ops][Task Manager] Adding integration test to ensure no `WorkloadAggregator` errors when there are unrecognized task types.

Fixes https://github.com/elastic/kibana-team/issues/1036

## Summary

Adding integration test as RCA action for incident where unrecognized task types was causing issues generating the workload portion of the task manager health report.

## To verify

Add this line to your code to that will throw an error when there are unrecognized task types when generating the health report

```
--- a/x-pack/plugins/task_manager/server/task_type_dictionary.ts
+++ b/x-pack/plugins/task_manager/server/task_type_dictionary.ts
@@ -128,6 +128,7 @@ export class TaskTypeDictionary {
   }

   public get(type: string): TaskDefinition | undefined {
+    this.ensureHas(type);
     return this.definitions.get(type);
   }
```

Run the integration test `node scripts/jest_integration.js x-pack/plugins/task_manager/server/integration_tests/removed_types.test.ts` and see that it fails because a `WorkloadAggregator` error is logged.
